src/com/google/javascript/jscomp/ProductionCoverageInstrumentationCallback.java

```diff
@@ -91,23 +115,99 @@ public void visit(NodeTraversal traversal, Node node, Node parent) {
+        }
+      }
+      if (!hasDefaultCase){
+        Node defaultBlock = IR.block();
+        defaultBlock.useSourceInfoIfMissingFromForTree(node);
+        Node defaultCase = IR.defaultCase(defaultBlock).useSourceInfoIfMissingFromForTree(node);
+        node.addChildToBack(defaultCase);
+        instrumentBlockNode(defaultBlock, sourceFileName, functionName, BRANCH_DEFAULT_TYPE);
+      }
+    } else if (NodeUtil.isLoopStructure(node)) {
+      Node blockNode = NodeUtil.getLoopCodeBlock(node);
+      checkNotNull(blockNode);
+      instrumentBlockNode(blockNode, sourceFileName, functionName, BRANCH_TYPE);
+
+      Node newNode =
+          newInstrumentationNode(blockNode, sourceFileName, functionName, BRANCH_DEFAULT_TYPE);
+      blockNode.getGrandparent().addChildAfter(newNode, blockNode.getParent());
+      compiler.reportChangeToEnclosingScope(blockNode.getParent());
+    } else if (node.isHook()) {
+      Node ifTernaryIsTrueExpression = node.getSecondChild();
+      Node ifTernaryIsFalseExpression = node.getLastChild();
+
+      addInstrumentationNodeWithComma(
+          ifTernaryIsTrueExpression, sourceFileName, functionName, BRANCH_TYPE);
+      addInstrumentationNodeWithComma(
+          ifTernaryIsFalseExpression, sourceFileName, functionName, BRANCH_TYPE);
+
+      compiler.reportChangeToEnclosingScope(node);
+    } else if (node.isOr() || node.isAnd() || node.isNullishCoalesce()) {
+      // Only instrument the second child of the binary operation because the first child will
+      // always execute, or the first child is part of a chain of binary operations and would have
+      // already been instrumented.
+      Node secondExpression = node.getLastChild();
+      addInstrumentationNodeWithComma(
+          secondExpression, sourceFileName, functionName, BRANCH_TYPE);
+
+      compiler.reportChangeToEnclosingScope(node);
     }
   }
 
   /**
-   * Iterate over all collected block nodes within a Script node and add a new child to the front of
-   * each block node which is the instrumentation Node
+   * Given a node, this function will create a new instrumentationNode and combine it with the
+   * original node using a COMMA node.
+   */
+  private void addInstrumentationNodeWithComma(
+      Node originalNode, String fileName, String functionName, String type) {
+    Node parentNode = originalNode.getParent();
+    parentNode.removeChild(originalNode);
+    Node newInstrumentationNode =
+        newInstrumentationNode(originalNode, fileName, functionName, type);
+
+    // newInstrumentationNode returns an EXPR_RESULT which cannot be a child of a COMMA node.
+    // Instead we use the child of of the newInstrumentatioNode which is a CALL node.
+    Node childOfInstrumentationNode = newInstrumentationNode.getFirstChild().detach();
+    Node infusedExp =
+        StatementFusion.fuseExpressionIntoExpression(childOfInstrumentationNode, originalNode);
+    parentNode.addChildToBack(infusedExp);
```

Comment:
Given the way I implemented it, yes this will always be the case. Although this is dependent on some behavior that isn't to obvious. I did some small changes by creating a clone of the original node and replacing the original node with the infusedNode to make it a but more clear.